File: SecondYear/Anantj2499_AnantJaiswal_2226cseml1088_2/Week_2/ReverseBits.java

    public class ReverseBits {
        public static long reverseBits(long n) {
            // Write your code here
            long[] arr = new long[32];
            long temp =n;
            int count = 0;
            while(temp>0){
                arr[count++]= temp%2;
                temp/=2;
            }
            for(int i =count; i<arr.length; i++){
                arr[i]=0;
            }
            temp =0;
            long index =1;
            for(int i=31;i>=0;i--){
                temp+=arr[i]*index;
                index*=2;
            }
            return temp;
        }
    }

File: SecondYear/Serve-er_SarveshBaranwal_2226cseAI&ml1045_2/Week_2/ReverseBits.cpp
  #include<Math.h>
  
  long reverseBits(long n) {
    // Write your code here.
    int arr[32]={0};
    int bit,cnt=32,sum=0;
    for(int i=0;i<32;i++){
          bit =n%2;
          arr[i]=bit;
          n=n/2;
    }
    long long int ans=0;
    int cnt=0;
    for(int i=31;i>=0;i--)
    {
        ans=ans+(arr[i]*pow(2,cnt));
        cnt++;
    }
    return ans;
  
}
